### PR TITLE
Do not use the deprecated pkg-resources package

### PR DESCRIPTION
**Pull request by avalentino**
Usage of pkg-resources is replaced with importlib.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -866,7 +867,7 @@ API Reference
                 self.config = self.parse_config(self.config_file)
                 self.asset_type = self.message[&#34;asset_type&#34;] if &#34;asset_type&#34; in self.message else &#34;&#34;
                 self.version = self.message[&#34;version&#34;] if &#34;version&#34; in self.message else &#34;&#34;
-                self.validator_version = pkg_resources.require(&#34;stac-validator&#34;)[0].version
+                self.validator_version = importlib.metadata.distribution(&#34;stac-validator&#34;)[0].version
```
Comment (jonhealy1) [inline docs/api.rst:870]: Hi. Is the [0] index here an error. I think.

**Review by jonhealy1 — CHANGES_REQUESTED** [review]
Thanks for doing this. One change I think. Also, could you add an entry to the CHAGNGELOG just under the Unreleased section. Thank you.

File: docs/api.rst (hunk 1 above)
Comment (avalentino) [reply docs/api.rst]: Should be fixed now.
Sorry for the mistake.

Comment (jonhealy1) [reply docs/api.rst]: Thanks no worries

Comment (jonhealy1) [reply docs/api.rst]: Getting this error in tests - 

```package = 'stac_check.lint'

    def _get_package(package) -> ModuleType:
        """Take a package name or module object and return the module.
    
        If a name, the module is imported.  If the passed or imported module
        object is not a package, raise an exception.
        """
        if hasattr(package, '__spec__'):
            if package.__spec__.submodule_search_locations is None:
                raise TypeError('{!r} is not a package'.format(
                    package.__spec__.name))
            else:
                return package
        else:
            module = import_module(package)
            if module.__spec__.submodule_search_locations is None:
>               raise TypeError('{!r} is not a package'.format(package))
E               TypeError: 'stac_check.lint' is not a package

/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/importlib/resources.py:49: TypeError

**Comment by avalentino** [comment]
Thanks @jonhealy1

**Pull request merged**